src/server/status_server.rs

```diff
@@ -89,11 +108,92 @@ impl StatusServer {
     }
 }
 
+// For handling fail points related requests
+#[cfg(not(feature = "no-fail"))]
+fn handle_fail_points_request(
+    req: Request<Body>,
+) -> Box<dyn Future<Item = Response<Body>, Error = hyper::Error> + Send> {
+    let path = req.uri().path().to_owned();
+    let method = req.method().to_owned();
+    let fail_path = format!("{}/", FAIL_POINTS_REQUEST_PATH);
+    let fail_path_has_sub_path: bool = path.starts_with(&fail_path);
+
+    match (method, fail_path_has_sub_path) {
```

Comment:
How about using actix-web to simplify the matching code? Hyper is too low-level and is not flexible enough for this high level task. @overvenus

Reply:
Ping @overvenus 

@BusyJay PTAL

Reply:
I think we can give actix-web a try in another PR.

Reply:
Should I try doing it? Not sure if this feature is urgent. I just started a new job and a bit busy these days. Might take a few weeks to make another actix-web PR ready 😄

Reply:
As suggested looks like you can start a new PR after this PR is being merged.

Reply:
Okay, I thought we want to close this PR and make a new one with actix-web. I don't mind merging this now as it is already working for most of the use cases, I suppose :) Let me know what else need to change here.

Reply:
You may wish to consider measuring the compile time impact of actix-web, I remember from some early versions the compile time impact was very large.